src/ProphecyTrait.php

```diff
@@ -46,20 +46,17 @@ protected function prophesize(?string $classOrInterface = null): ObjectProphecy
         if ($isPhpUnit9 && \is_string($classOrInterface)) {
             \assert($this instanceof TestCase);
             $this->recordDoubledType($classOrInterface);
-        }
-
-        return $this->getProphet()->prophesize($classOrInterface);
-    }
-
-    protected function setUp(): void
-    {
-        if (class_exists(Event::class)) {
+        } else {
             // PHPUnit 10.1+ compat
-            $this->registerFailureInterface(PredictionException::class);
-            $this->registerFailureInterface(UnexpectedCallException::class);
+            static $alreadyRegistered = false;
```

Comment:
The state should not be static, as different tests in the same class will still need to register it. This should be an instance property.

Reply:
No longer needed at all thanks to https://github.com/sebastianbergmann/phpunit/commit/a249d880f0d56f8fe9a8a8d2763406c9decbe521